### PR TITLE
ACS-4799: Elasticsearch query feature expansion - boosts

### PR DESCRIPTION
**Pull request by krdabrowski**
E2E tests verifying search boosts functionality for ElasticSearch.
https://alfresco.atlassian.net/browse/ACS-4799

### CODE REVIEW

File: tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchBoostedSearchTests.java [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+        STEP("Search for files and folders by name or creation time range with higher priority for name filter");
+        String boostedQuery = "TYPE:('cm:content'^3 OR 'cm:folder') AND (cm:name:" + SEARCH_TERM + "^4 OR cm:created:['" + timeFrom + "' TO '" + timeTo + "']^0.1)";
+        SearchRequest searchRequest = req("afts", boostedQuery);
+        searchQueryService.expectResultsInOrder(searchRequest, testUser, fileWithTermInName.getName(), folderWithTermInName.getName(), fileWithTermInTitle.getName(), folderWithTermInTitle.getName());
+
+        STEP("Search for files and folders by name or creation time range with higher priority for creation time range filter");
+        String invertedBoost = "TYPE:('cm:content'^3 OR 'cm:folder') AND (cm:name:" + SEARCH_TERM + "^0.1 OR cm:created:['" + timeFrom + "' TO '" + timeTo + "']^4)";
+        searchRequest = req("afts", invertedBoost);
+        searchQueryService.expectResultsInOrder(searchRequest, testUser, fileWithTermInTitle.getName(), folderWithTermInTitle.getName(), fileWithTermInName.getName(), folderWithTermInName.getName());
+    }
```
Comment (azakrzewski-hy) [inline tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchBoostedSearchTests.java:221]: Would it make sens to mark folder whit higher priority to make cross check of the search ? Files appear first before folder in preconditions

File: tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchBoostedSearchTests.java [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+
+    /**
+     * Verify if boosts works fine with words range search. Files containing words from alphabetical range (from 'mountain' to 'phrase', this includes word 'other') should be returned.
+     */
+    @Test(groups = { TestGroup.SEARCH })
+    public void testAftsQuery_wordsRangeSearchBoost()
+    {
+        STEP("Search for files and folders by name or words in content from given range with higher priority for name filter");
```
Comment (azakrzewski-hy) [inline tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchBoostedSearchTests.java:229]: Not sure about "and folders" in this description

File: tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchBoostedSearchTests.java [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+        String boostedQuery = "TYPE:'cm:content' AND (cm:name:" + SEARCH_TERM + "^3 OR cm:content:" + SEARCH_TERM + "..phrase^0.1)";
+        SearchRequest searchRequest = req("afts", boostedQuery);
+        searchQueryService.expectResultsStartingWithOneOf(searchRequest, testUser, fileWithTermInName.getName());
+        searchQueryService.expectResultsFromQuery(searchRequest, testUser, fileWithTermInName.getName(), fileWithPhraseInContent.getName(), fileWithDifferentTermInName.getName());
+
+        STEP("Search for files and folders by name or words in content from given range with higher priority for words in content from given range filter");
+        String invertedBoost = "TYPE:'cm:content' AND (cm:name:" + SEARCH_TERM + "^0.1 OR cm:content:" + SEARCH_TERM + "..phrase^3)";
+        searchRequest = req("afts", invertedBoost);
+        searchQueryService.expectResultsStartingWithOneOf(searchRequest, testUser, fileWithPhraseInContent.getName(), fileWithDifferentTermInName.getName());
+        searchQueryService.expectResultsFromQuery(searchRequest, testUser, fileWithTermInName.getName(), fileWithPhraseInContent.getName(), fileWithDifferentTermInName.getName());
```
Comment (azakrzewski-hy) [inline tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchBoostedSearchTests.java:239]: Does priority have no impact on expectResultsFromQuery() results? One step above we get the same

**Review by azakrzewski-hy — APPROVED** [review]

**Review by mpichura — APPROVED** [review]
LGTM

File: tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchBoostedSearchTests.java (hunk 1 above)
Comment (krdabrowski) [reply tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchBoostedSearchTests.java:221]: No problem, switched priority from file to folder

File: tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchBoostedSearchTests.java (hunk 2 above)
Comment (krdabrowski) [reply tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchBoostedSearchTests.java]: Good catch! Removed copy-paste leftover

**Comment by krdabrowski** [comment]
One intermittent test fail occurred:

Error:  Tests run: 163, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 3,384.69 s <<< FAILURE! - in TestSuite
Error:  testSiteUseCasesForCreateModifyDeleteSite(org.alfresco.elasticsearch.reindexing.ElasticsearchSiteIndexingTests)  Time elapsed: 85.033 s  <<< FAILURE!
java.lang.AssertionError: Maximum retry period reached, test failed.
	at org.alfresco.elasticsearch.reindexing.ElasticsearchSiteIndexingTests.assertSiteQueryResult(ElasticsearchSiteIndexingTests.java:333)
	at org.alfresco.elasticsearch.reindexing.ElasticsearchSiteIndexingTests.testSiteUseCasesForCreateModifyDeleteSite(ElasticsearchSiteIndexingTests.java:210)
Caused by: java.lang.AssertionError: Unexpected search results - got [documentLibrary, EsSiteTesttest1-bb2b48c3-e6aa-4994-9e26-427413445e52.txt] expected [documentLibrary, EsSiteTesttest1-bb2b48c3-e6aa-4994-9e26-427413445e52.txt, EsSiteTesttest2-68d0ebfc-ecc4-43fd-bfaa-658b2e7f89e7.txt]
	at org.alfresco.elasticsearch.reindexing.ElasticsearchSiteIndexingTests.assertSiteQueryResult(ElasticsearchSiteIndexingTests.java:333)
	at org.alfresco.elasticsearch.reindexing.ElasticsearchSiteIndexingTests.testSiteUseCasesForCreateModifyDeleteSite(ElasticsearchSiteIndexingTests.java:210)
[INFO] 
[INFO] Results:
[INFO] 
Error:  Failures: 
Error:    ElasticsearchSiteIndexingTests>AbstractTestNGSpringContextTests.run:186->testSiteUseCasesForCreateModifyDeleteSite:210->assertSiteQueryResult:333 Maximum retry period reached, test failed.
Rerunning the test

**Pull request merged**